src/main/java/com/snowflake/kafka/connector/internal/BufferThreshold.java

```diff
@@ -0,0 +1,213 @@
+    this.bufferKafkaRecordCountThreshold = bufferKafkaRecordCountThreshold;
+  }
+
+  /**
+   * Returns true if size of current buffer is more than threshold provided (Both in bytes).
+   *
+   * <p>Threshold is config parameter: {@link
+   * com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig#BUFFER_SIZE_BYTES}
+   *
+   * @param currentBufferSizeInBytes current size of buffer in bytes
+   * @return true if bytes threshold has reached.
+   */
+  public boolean isFlushBufferedBytesBased(final long currentBufferSizeInBytes) {
+    return currentBufferSizeInBytes >= bufferSizeThresholdBytes;
+  }
+
+  /**
+   * Returns true if number of ({@link SinkRecord})s in current buffer is more than threshold
+   * provided.
+   *
+   * <p>Threshold is config parameter: {@link
+   * com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig#BUFFER_COUNT_RECORDS}
+   *
+   * @param currentBufferedRecordCount current size of buffer in terms of number of kafka records
+   * @return true if number of kafka threshold has reached in buffer.
+   */
+  public boolean isFlushBufferedRecordCountBased(final long currentBufferedRecordCount) {
+    return currentBufferedRecordCount != 0
+        && currentBufferedRecordCount >= bufferKafkaRecordCountThreshold;
+  }
+
+  /**
+   * If difference between current time and previous flush time is more than threshold return true.
+   *
+   * @param previousFlushTimeStampMs when were previous buffered records were flushed into internal
+   *     stage for snowpipe based implementation or previous buffered records were sent in
+   *     insertRows API of Streaming Snowpipe.
+   * @return true if time based threshold has reached.
+   */
+  public boolean isFlushTimeBased(final long previousFlushTimeStampMs) {
+    final long currentTimeMs = System.currentTimeMillis();
+    return (currentTimeMs - previousFlushTimeStampMs)
+        >= (this.flushTimeThresholdSeconds * SECOND_TO_MILLIS);
+  }
+
+  /** Get flush time threshold in seconds */
+  public long getFlushTimeThresholdSeconds() {
+    return flushTimeThresholdSeconds;
+  }
+
+  /**
+   * Check if provided snowflake kafka connector buffer properties are within permissible values
+   *
+   * @param providedSFConnectorConfig provided by customer
+   * @return true if valid.
+   */
+  public static boolean validateBufferThreshold(
+      Map<String, String> providedSFConnectorConfig, IngestionMethodConfig ingestionMethodConfig) {
+    if (!providedSFConnectorConfig.containsKey(
+        SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC)) {
+      LOGGER.error(
+          Logging.logMessage("{} is empty", SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
+      return false;
+    } else {
+      try {
+        long time =
+            Long.parseLong(
+                providedSFConnectorConfig.get(SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
+        if (ingestionMethodConfig.equals(IngestionMethodConfig.SNOWPIPE)) {
```

Comment:
To consolidate code what you could do is something like this:
```
if (ingestionMethodConfig.equals(IngestionMethodConfig.SNOWPIPE)) {
    comparisonTime = BUFFER_FLUSH_TIME_SEC_MIN;
    configOption = SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC_MIN;
} else {
    comparisonTime = STREAMING_BUFFER_FLUSH_TIME_MINIMUM_SEC;
    configOption = SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC,
}
if (time < comparisonTime) {
            LOGGER.error(
                (Logging.logMessage(
                    "{} is {}, it should be greater than {}",
                    configOption
                    time,
                    comparisonTime)));
            return false;
}                
```

Reply:
Will do!